### PR TITLE
add support for do_not_archive attribute and for plugin CHAIN_POSITION...

### PR DESCRIPTION
**Pull request by galgeek**
... to better support a plugin setting do_not_archive, as for a plugin detecting facebook captchas

### CODE REVIEW

File: warcprox/writerthread.py [hunk 1]
```diff
@@ -81,8 +81,12 @@ def _should_archive(self, recorded_url):
                   if recorded_url.warcprox_meta
                      and 'warc-prefix' in recorded_url.warcprox_meta
                   else self.options.prefix)
+        do_not_archive = (recorded_url.do_not_archive
+                          if recorded_url.do_not_archive
+                          else False)
         # special warc name prefix '-' means "don't archive"
-        return prefix != '-' and self._filter_accepts(recorded_url)
+        return (prefix != '-' and (not do_not_archive)
+                and self._filter_accepts(recorded_url))
```
Comment (nlevitt) [inline warcprox/writerthread.py]: Let's get rid of the local variable `do_not_archive` ☝🏻and write this like so:
```
return (prefix != '-' and not recorded_url.do_not_archive
        and self._filter_accepts(recorded_url))
```

File: warcprox/writerthread.py [hunk 2]
```diff
@@ -81,8 +81,12 @@ def _should_archive(self, recorded_url):
                   if recorded_url.warcprox_meta
                      and 'warc-prefix' in recorded_url.warcprox_meta
                   else self.options.prefix)
+        do_not_archive = (recorded_url.do_not_archive
+                          if recorded_url.do_not_archive
+                          else False)
```
Comment (nlevitt) [inline warcprox/writerthread.py]: What's the purpose of this local variable?

File: warcprox/writerthread.py (hunk 1 above)
Comment (galgeek) [reply warcprox/writerthread.py]: agreed

**Comment by nlevitt** [comment]
Do you mind adding a test case for this?

**Comment by galgeek** [comment]
Sure, I can add a test case.

File: warcprox/version.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+1.4-20160105052702-f79e744
```
Comment (nlevitt) [inline warcprox/version.txt]: Accidental commit?

Comment (galgeek) [reply warcprox/version.txt]: Yes, indeed.

**Pull request merged**